pkg/storage/etcd3/store.go

```diff
@@ -0,0 +1,413 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd3
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"path"
+	"reflect"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/conversion"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+	"k8s.io/kubernetes/pkg/storage/etcd"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+)
+
+type store struct {
+	client     *clientv3.Client
+	codec      runtime.Codec
+	versioner  storage.Versioner
+	pathPrefix string
+}
+
+type elemForDecode struct {
+	data []byte
+	rev  uint64
+}
+
+type objState struct {
+	obj  runtime.Object
+	meta *storage.ResponseMeta
+	rev  int64
+	data []byte
+}
+
+func newStore(c *clientv3.Client, codec runtime.Codec, prefix string) *store {
+	return &store{
+		client:     c,
+		versioner:  etcd.APIObjectVersioner{},
+		codec:      codec,
+		pathPrefix: prefix,
+	}
+}
+
+func (s *store) Backends(ctx context.Context) []string {
```

Comment:
Those are implementation of storage.Interface. It's not really a convention to put comments on implementation. But sure, I will put some simple thing to say that they "implements storage.Interface"